### DISCUSSION TITLE
Netbeans 18 (and 17) ignoring PHP Xdebug port configuration

### DISCUSSION THREAD

**Discussion by dnessett** [apache/netbeans#6273] [General]
I am experiencing the ubiquitous "Netbeans won't connect to Xdebug" problem. I have extensively googled this problem and have not found a solution. I then did some local debugging and discovered that Netbeans is ignoring the port configured for debugging in its PHP configuration panel. I have attached three screen shots showing that this is true. The first shows the PHP configuration for netbeans. The second shows the partial configuration for Xdebug (3.2.2) that sets the port Xdebug listens on (I used xdebuginfo() for this information). The third shows a Wireshark capture that demonstrates that netbeans is using port 80 to attempt to start Xdebug, rather than port 9003, which is the configured value. I am using netbeans 18, but the problem also existed for netbeans 17 (I upgraded hoping netbeans 18 had a bug fix for the problem). My configuration is: Mac OS 13.4.1, Netbeans 18, Java: 17.0.7; OpenJDK 64-Bit Server VM 17.0.7+7, PHP 8.1.20, Xdebug 3.2.2.

I considered reporting this as a bug, but could not figure out a way to reproduce the bug on another machine. I can try to run debugging of netbeans on my machine, but am not sure if I can debug netbeans using netbeans. I also have Visual Studio installed on my machine, but have never used it to debug anything. So, if someone can help me to debug the problem on my machine I am willing to do that and report back my results.

![Netbeans 18 PHP configuration](https://github.com/apache/netbeans/assets/42894619/77dc608f-0359-4d98-b68b-10ef681c5460)

![XDEBUG 3 2 2 configuraition](https://github.com/apache/netbeans/assets/42894619/ffce2d11-7c14-47dc-8e79-22198cd2fa7c)

![Wireshark packet capture](https://github.com/apache/netbeans/assets/42894619/f823e31c-870a-470a-99fd-dbeb333035de)

**Comment by troizet** [comment]
The debug session is started by `Xdebug` when the php script is executed, not `ide`. See https://xdebug.org/docs/dbgp#initiating-a-debugging-session
`Netbeans` when you click on `Debug project` makes a request to open the debug page in the browser (this is the request that shows up in `Wireshark`) and waits for `Xdebug` to initialize the debug session.
The request is configured in `Run -> Set Project Configuration` or under `Run Configurations` in the project properties. In `Advanced...` you can specify `Do not Open Web Browser` to disable this request. In this case, the debugging php script should be run manually after clicking `Debug project`.

**Reply by dnessett** [reply]
Thanks for your responses, troizet. I have resolved the problem and can now close this discussion.

**Comment by troizet** [comment]
To debug `Netbeans` in `Netbeans` you must first build `Netbeans` from sources, as specified here: https://github.com/apache/netbeans#building-netbeans.
Then open the module you want to debug as a regular project (`File -> Open project`). In this case it will be the [`php.dbgp`](https://github.com/apache/netbeans/tree/master/php/php.dbgp) module.

**Comment by dnessett** [comment]
Problem resolved.
